src/DebuggerData.cpp

```diff
@@ -309,13 +315,15 @@ bool Breakpoints::isWatchpoint(quint16 addr, QString* id, bool checkSlot)
 	return false;
 }
 
-int Breakpoints::findBreakpoint(quint16 addr)
+std::optional<uint16_t>
+Breakpoints::findBreakpoint(uint16_t addr)
 {
-	if (auto i = ranges::lower_bound(breakpoints, addr, {}, &Breakpoint::address);
-	    i != breakpoints.end() && i->address == addr) {
+        if (auto i = ranges::lower_bound(breakpoints, addr,
+            [](auto& bp1, auto& addr) { return bp1.start < addr; }, &Breakpoint::range);
```

Comment:
detail: currently this uses a weird mix of a custom comparison function with a simple projection function. Also the parameter name `bp1` is misleading: it's now a range, not a breakpoint.

Consider choosing one of these alternatives:

    // alternative 1: do everything in the projection function
    if (auto i = ranges::lower_bound(breakpoints, addr, {}, [](const auto& bp) { return bp.range.start; }) {
    // alternative 2: do everything in the comparison function
    if (auto i = ranges::lower_bound(breakpoints, addr, [](auto& bp, auto addr) { return bp.range.start < addr; }) {